rs/artifact_pool/src/certification_pool.rs

```diff
@@ -143,35 +199,77 @@ impl CertificationPoolImpl {
     }
 
     fn update_metrics(&self) {
-        // Validated artifacts metrics
-        self.validated_pool_metrics
-            .pool_artifacts
-            .with_label_values(&[CERTIFICATION_ARTIFACT_TYPE])
-            .set(self.validated.certifications().size() as i64);
-        self.validated_pool_metrics
-            .pool_artifacts
-            .with_label_values(&[CERTIFICATION_SHARE_ARTIFACT_TYPE])
-            .set(self.validated.certification_shares().size() as i64);
-
-        // Unvalidated artifacts metrics
+        // Unvalidated section
+        if let (Some(min_cert), Some(max_cert), Some(min_share), Some(max_share)) = (
+            self.unvalidated_cert_index.range(..).next(),
+            self.unvalidated_cert_index.range(..).last(),
+            self.unvalidated_share_index.range(..).next(),
+            self.unvalidated_share_index.range(..).last(),
+        ) {
+            self.unvalidated_pool_metrics
+                .certification
+                .min_height
+                .set(min_cert.0.get() as i64);
+            self.unvalidated_pool_metrics
+                .certification
+                .max_height
+                .set(max_cert.0.get() as i64);
+            self.unvalidated_pool_metrics
+                .certification_share
+                .min_height
+                .set(min_share.0.get() as i64);
+            self.unvalidated_pool_metrics
+                .certification_share
+                .max_height
+                .set(max_share.0.get() as i64);
+        }
         self.unvalidated_pool_metrics
-            .pool_artifacts
-            .with_label_values(&[CERTIFICATION_ARTIFACT_TYPE])
+            .certification
+            .count
             .set(self.unvalidated_cert_index.size() as i64);
         self.unvalidated_pool_metrics
-            .pool_artifacts
-            .with_label_values(&[CERTIFICATION_SHARE_ARTIFACT_TYPE])
+            .certification_share
+            .count
             .set(self.unvalidated_share_index.size() as i64);
+
+        // Validated section
+        if let (Some(cert), Some(share)) = (
+            self.validated.certifications().height_range(),
+            self.validated.certification_shares().height_range(),
+        ) {
```

Comment:
(The same comment applies to the `if let` above)

Don't we want to split this into two `if let`s?

```
If let Some(cert) = self.validated.certifications().height_range() {
   ...
}
If let Some(share) = self.validated.certification_shares().height_range() {
   ...
}
```
right now we won't report any metric if we either don't have shares or certifications in the pool